src/pages/MoodStatistics/AgencyStatisticsSummary.tsx

```diff
@@ -1,30 +1,91 @@
 import { Grid, Hidden, Typography } from '@mui/material'
 import ProgressPieChart from '../../components/ProgressPieChart'
-import React from 'react'
+import React, { useEffect } from 'react'
 import {
   averageFeeling,
   calculateCheer,
   getDataSet,
   getTotalDataSet,
 } from '../../utils/feelingUtils'
-import { useSelector } from 'react-redux'
+import { useDispatch, useSelector } from 'react-redux'
 import { useTranslation } from 'react-i18next'
+import {
+  fetchContractsAsAgency,
+  fetchEmploymentContractsAsWorkerOrBusiness,
+} from '../../actions/contractActions'
+import { IRootState } from '../../utils/store'
+import { roles } from '../../types/types'
+
+type Feeling = {
+  comment: string
+  createdAt: string
+  feeling: number
+  worker: string
+  __v: number
+  _id: string
+}
 
 /**
  * @component
  */
-const AgencyStatisticsSummary: React.FC<any> = () => {
-  const { feelings } = useSelector((state: any) => state.feeling)
-  const feelingsList = []
+const AgencyStatisticsSummary: React.FC = () => {
+  const { data } = useSelector((state: IRootState) => state.user)
+  const role = data.role //Current role: Agency or Business
+
+  const workersList: string[] = [] //IDs of all workers associated with Business or Agency
+  const allFeelings: Feeling[] = [] //All feelings
+  const feelingsList: number[] = [] //Feelings of only workers associated with the Business/Agency as numeric values
+  const dispatch = useDispatch()
+
+  if (role == roles.Agency) {
```

Comment:
```suggestion
  if (role === roles.Agency) {
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality